packages/langium/src/service-registry.ts

```diff
@@ -4,38 +4,41 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-import type { LangiumServices } from './services.js';
+import type { LangiumCoreServices } from './services.js';
 import { UriUtils, type URI } from './utils/uri-util.js';
 
 /**
- * The service registry provides access to the language-specific services. These are resolved
+ * The core service registry provides access to the language-specific core services. These are resolved
  * via the URI of a text document.
  */
-export interface ServiceRegistry {
+export interface ServiceRegistry<T extends LangiumCoreServices> {
```

Comment:
AFAICS this type parameter is not necessary, Typescript allows to formulate the hand full of relevant occurrences differently 🙂

Reply:
It's there to ensure that anything implementing the ServiceRegistry interface works with at least the core service set or greater. Although you can drop it & the `DefaultServicerRegistry` constraints will suffice here, you lose any typechecking that would flag a service registry violating this assumption (like a user customized variant using that interface).

Reply:
dropped.